server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailQueryMethodContract.scala

```diff
@@ -267,57 +271,115 @@ trait EmailQueryMethodContract {
+        .post.prettyPeek()
       .`then`
         .statusCode(SC_OK)
         .contentType(JSON)
         .extract
         .body
         .asString
 
-      val responseSecondCall = `given`
+    assertThatJson(response)
+      .inPath("$.methodResponses[0][1].ids")
+      .isEqualTo(s"""["${messageId3.serialize()}", "${messageId2.serialize()}", "${messageId1.serialize()}"]""")
+    }
+  }
+
+  @Test
+  def listMailsShouldBeSortedASCByArrivalAtWithComparator(server: GuiceJamesServer): Unit = {
+    val message: Message = Message.Builder
+      .of
+      .setSubject("test")
+      .setBody("testmail", StandardCharsets.UTF_8)
+      .build
+    server.getProbe(classOf[MailboxProbeImpl]).createMailbox(MailboxPath.inbox(BOB))
+    val otherMailboxPath = MailboxPath.forUser(BOB, "other")
+    server.getProbe(classOf[MailboxProbeImpl]).createMailbox(otherMailboxPath)
+    val messageId1 = server.getProbe(classOf[MailboxProbeImpl])
+      .appendMessage(BOB.asString, MailboxPath.inbox(BOB), AppendCommand.from(message))
+      .getMessageId
+    Thread.sleep(1000) // To enforce receivedAt ordering
+    val messageId2 = server.getProbe(classOf[MailboxProbeImpl])
+      .appendMessage(BOB.asString, MailboxPath.inbox(BOB), AppendCommand.from(message))
+      .getMessageId
+    Thread.sleep(1000) // To enforce receivedAt ordering
+    val messageId3 = server.getProbe(classOf[MailboxProbeImpl])
+      .appendMessage(BOB.asString, MailboxPath.inbox(BOB), AppendCommand.from(message))
+      .getMessageId
+    Thread.sleep(1000) // To enforce receivedAt ordering
+    val request =
+      s"""{
+         |  "using": [
+         |    "urn:ietf:params:jmap:core",
+         |    "urn:ietf:params:jmap:mail"],
+         |  "methodCalls": [[
+         |    "Email/query",
+         |    {
+         |      "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+         |      "comparator": [{
+         |        "property":"receivedAt",
+         |        "isAscending":"true",
+         |        "collation":"some thing not matter and ignored"
+         |      }]
+         |    },
+         |    "c1"]]
+         |}""".stripMargin
+
+    awaitAtMostTenSeconds.untilAsserted { () =>
+      val response = `given`
         .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
         .body(request)
       .when
-        .post
+        .post.prettyPeek()
       .`then`
         .statusCode(SC_OK)
         .contentType(JSON)
         .extract
         .body
         .asString
 
-      assertThatJson(responseFirstCall).isEqualTo(responseSecondCall)
+    assertThatJson(response)
+      .inPath("$.methodResponses[0][1].ids")
+      .isEqualTo(s"""["${messageId3.serialize()}", "${messageId2.serialize()}", "${messageId1.serialize()}"]""")
     }
   }
 
   @Test
-  def shouldListMailsInASpecificUserMailboxes(server: GuiceJamesServer): Unit = {
+  def listMailsShouldBeReturnErrorBySortWithInvalidComparator(server: GuiceJamesServer): Unit = {
```

Comment:
i don't understand should we return an error. Or sort like if no compartor is specified ?

Reply:
I should give it better name.
This test for `Comparator` with missing required property `receivedAt` so i suppose to be return error by server?

Reply:
```suggestion
  def listMailsShouldReturnErrorWhenPropertyFieldInComparatorIsInvalid(server: GuiceJamesServer): Unit = {
```